From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Thu, 24 Apr 2025 23:39:21 +0200
Subject: [PATCH] Adjust tests for libxml 2.14

It no longer considers undeclared entities as errors when parsing
content fragments, and I think this is expected behavior.

See: https://gitlab.gnome.org/GNOME/libxml2/-/issues/895
---
 src/test/regress/expected/xml.out | 14 ++++++--------
 1 file changed, 6 insertions(+), 8 deletions(-)

diff --git a/src/test/regress/expected/xml.out b/src/test/regress/expected/xml.out
index bd548f2bc5ba..54db338ae8bf 100644
--- a/src/test/regress/expected/xml.out
+++ b/src/test/regress/expected/xml.out
@@ -255,28 +255,26 @@ DETAIL:  line 1: xmlParseEntityRef: no name
 <invalidentity>&</invalidentity>
                 ^
 SELECT xmlparse(content '<undefinedentity>&idontexist;</undefinedentity>');
-ERROR:  invalid XML content
-DETAIL:  line 1: Entity 'idontexist' not defined
-<undefinedentity>&idontexist;</undefinedentity>
-                             ^
+                    xmlparse                     
+-------------------------------------------------
+ <undefinedentity>&idontexist;</undefinedentity>
+(1 row)
+
 SELECT xmlparse(content '<invalidns xmlns=''&lt;''/>');
          xmlparse          
 ---------------------------
  <invalidns xmlns='&lt;'/>
 (1 row)
 
 SELECT xmlparse(content '<relativens xmlns=''relative''/>');
             xmlparse            
 --------------------------------
  <relativens xmlns='relative'/>
 (1 row)
 
 SELECT xmlparse(content '<twoerrors>&idontexist;</unbalanced>');
 ERROR:  invalid XML content
-DETAIL:  line 1: Entity 'idontexist' not defined
-<twoerrors>&idontexist;</unbalanced>
-                       ^
-line 1: Opening and ending tag mismatch: twoerrors line 1 and unbalanced
+DETAIL:  line 1: Opening and ending tag mismatch: twoerrors line 1 and unbalanced
 <twoerrors>&idontexist;</unbalanced>
                                     ^
 SELECT xmlparse(content '<nosuchprefix:tag/>');
